A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,405 @@
+    plugins such that metrics for only certain channels are recorded on those
+    plugins. (Current implementations only support global registration of stats
+    plugins.) There can be multiple stats plugins registered as well.
+*   gRPC/Third-party plugins control the metric instrument definition - As gRPC
+    library/component writers, we want to be in control of the metric definition
+    that gRPC provides, irrespective of the stats plugin used. OpenTelemetry is
+    the only supported stats plugin as of now, but that might change in the
+    future. Third-party plugins to gRPC that choose to use this architecture
+    would be responsible for the definitions of metrics exported by their
+    plugins.
+*   Stats plugins should be able to enable/disable recorded metrics - gRPC has
+    many components that would want to record metrics. Not all of those metrics
+    will be interesting to stats plugins. Stats plugin writers should allow
+    users to configure which metrics and enabled/disabled.
+
+### Proposed Architecture
+
+#### Global Instruments Registry
+
+A new `GlobalInstrumentsRegistry` is added. Any metric instrument that needs to
+be recorded has to register its descriptor to this registry.
+
+Instrument Descriptor -
+
+*   Name (string) - The name of the instrument
+*   Description (string) - The description of the instrument
+*   Unit (string) - The unit for the recordings, e.g. `s` (seconds), `By`
+    (bytes).
+*   Type - Implementation defined instrument type, e.g. Long Counter, Double
+    Histogram.
+*   Set of Label Keys (set of strings) - These determine the set of labels that
+    will be present with each recording/measurement.
+*   Set of Optional Label Keys (set of strings) - This is an additional set of
+    labels that stats plugins can choose to either record or throw away based on
+    user configuration. This is useful in cases where certain labels might be
+    deemed not useful for the majority of the cases, and hence shouldn't be
+    recorded by default. (OpenTelemetry allows users to create custom views that
+    can filter labels, but in the absence of custom views, the view takes the
+    entire set of labels provided by the instrumentation library.)
+
+On registering an instrument a `GlobalInstrumentHandle` is returned. The
+suggested implementation of this handle is an index that allows stats plugins to
+quickly look up the corresponding instrument when a measurement is being
+recorded.
+
+When registering an instrument, a hint is provided as to whether stats plugins
+should consider that instrument on-by-default or off-by-default. The rationale
+being that component/plugin writers would have the best judgment on whether a
+metric should be enabled by default or not, especially if collecting the
+measurement is expensive.
+
+Instrument descriptors are expected to be registered at static initialization
+time (initialization before main). Implementations can optionally provide the
```

Comment:
I don't think we should say that implementations can support dynamic registration, because we haven't designed that yet.  Instead, let's say that we may decide to add this capability in the future, but it's not part of the design for now.